src/main/java/net/starlark/java/eval/StarlarkInt.java

```diff
@@ -446,7 +464,13 @@ public static int compare(StarlarkInt x, StarlarkInt y) {
       /* fall through */
     }
 
-    return x.toBigInteger().compareTo(y.toBigInteger());
+    int xo = x.orderOfMagnitude();
```

Comment:
Delete LL457-459? (GitHub won't let me comment on that line.)

The invariant at this point (L467) is that at least one of the operands is Big. Can we do the comparison efficiently without adding the new virtual method, something like this:

```
public static int compare(StarlarkInt x, StarlarkInt y) {
  long xl, yl;
  boolean xbig, ybig;
  try {
    xl = x.toLongFast();
  } catch (Overflow) {
    xbig = true;
  }
  try {
    yl = y.toLongFast();
  } catch (Overflow) {
    ybig = true;
  }

  // If only one operand is big, its magnitude is greater than the other operand,
  // which can be ignored (treated as zero).
  if (xbig) {
    return ybig ?  ((Big) x).compareTo(y) : x.signum();
  } else {
    return ybig ?  -y.signum(), Long.compare(x, y);
  }
}
```